src/Dlp/V2beta1/Gapic/DlpServiceGapicClient.php

```diff
@@ -183,13 +166,54 @@ private static function getLongRunningDescriptors()
 
     private static function getGapicVersion()
     {
-        if (file_exists(__DIR__.'/../VERSION')) {
-            return trim(file_get_contents(__DIR__.'/../VERSION'));
-        } elseif (class_exists('\Google\Cloud\ServiceBuilder')) {
-            return \Google\Cloud\ServiceBuilder::VERSION;
-        } else {
-            return;
+        if (!self::$gapicVersionLoaded) {
+            if (file_exists(__DIR__.'/../VERSION')) {
+                self::$gapicVersion = trim(file_get_contents(__DIR__.'/../VERSION'));
+            } elseif (class_exists('\Google\Cloud\Version')) {
+                self::$gapicVersion = \Google\Cloud\Version::VERSION;
+            }
+            self::$gapicVersionLoaded = true;
+        }
+
+        return self::$gapicVersion;
+    }
+
+    /**
+     * Formats a string containing the fully-qualified path to represent
+     * a result resource.
+     *
+     * @param string $result
+     *
+     * @return string The formatted result resource.
+     * @experimental
+     */
+    public static function resultName($result)
+    {
+        return self::getResultNameTemplate()->render([
+            'result' => $result,
+        ]);
+    }
+
+    /**
+     * Parses a formatted name string and returns an associative array of the components in the name.
+     * The following name formats are supported:
+     * - inspect/results/{result}.
+     *
+     * @param string $formattedName The formatted name string
+     *
+     * @return array An associative array from name component IDs to component values.
+     * @experimental
+     */
+    public static function parseName($formattedName)
```

Comment:
That could be useful, but right now we don't expose the templates. Perhaps we can add this in future if we see a need?

Reply:
I am making the argument I see the need right now. We are paying a cost here, which is to loop over items that we know won't successfully evaluate.

The previous iteration allowed for users to select exactly what template they wanted (through choosing the correct method), I feel we could still retain that as it is valuable. What about a map?

Roughly...
```php
function parseName($formattedName, $template = null)
{
    // this would be instantiated elsewhere...
    $templates = [
        'result' => new PathTemplate('inspect/results/{result}')
    ];

    if (isset($templates[$template]) {
        return $templates[$template]->match($formattedName);
    }

    // otherwise, if $template === null, do what we are doing now
}

parseName('inspect/results/my-result', 'result');
```

I am okay with thinking through this more later, so as to not block.

Reply:
I like it, and I think it would be fairly easy to implement. Let me take a look now and see what I can do